torchao/experimental/ops/mps/test/test_lowbit.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import torchao_mps_ops
+import unittest
+
+
+def parameterized(test_cases):
+    def decorator(func):
+        def wrapper(self):
+            for case in test_cases:
+                with self.subTest(case=case):
+                    func(self, *case)
+
+        return wrapper
+
+    return decorator
+
+
+class TestLowBitQuantWeightsLinear(unittest.TestCase):
+    cases = [
+        (nbit, *param)
+        for nbit in range(1, 8)
+        for param in [
+            (1, 32, 32, 32),
+            (128, 48, 64, 32),
+            (17, 1024, 512, 256),
+        ]
+    ]
+
+    def _init_tensors(self, group_size, M, K, N, nbit, device="mps"):
+        max_abs = 1 << (nbit - 1)
+        ceil_K_group_size = (K + group_size - 1) // group_size
```

Comment:
This is making it sound like you can handle any groupsize but tests dont account for it

Reply:
We could handle any groupsize. But in practice we only handle the ones instantiated in the shaders: 32, 64, 128, 256. This also agrees with PyTorch's int4 implementation